tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,443 @@ int main(int argc, char **argv)
+                    0x00,
+                    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                };
+
+                uint8_t count = sizeof(ecdhe_rsa_chacha20_boosted_wire) / S2N_SSLv2_CIPHER_SUITE_LEN;
+
+                if (s2n_chacha20_poly1305.is_available()) {
+                    EXPECT_SUCCESS(s2n_set_cipher_as_sslv2_server(connection, ecdhe_rsa_chacha20_boosted_wire, count));
+                    EXPECT_EQUAL(connection->secure->cipher_suite, &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256);
+                }
+
+                if (!s2n_chacha20_poly1305.is_available()) {
+                    EXPECT_FAILURE_WITH_ERRNO(s2n_set_cipher_as_sslv2_server(connection, ecdhe_rsa_chacha20_boosted_wire, count), S2N_ERR_CIPHER_NOT_SUPPORTED);
+                }
+            }
+        }
+
+        /* Test ChaCha20 boosting behaviour for tls 1.2 connection w/ that supports only rsa certs */ 
+        {
+            static struct s2n_cipher_suite *cipher_suites_with_tls12_chacha20[] = {
+                    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       /* 0xC0,0x23 */
+                    &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256, /* 0xCC,0xA9 */
+                    &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   /* 0xCC,0xA8 */
+                    &s2n_dhe_rsa_with_chacha20_poly1305_sha256,     /* 0xCC,0xAA */
+            };
+
+            struct s2n_cipher_preferences cipher_preferences = (struct s2n_cipher_preferences) {
+                    .count = s2n_array_len(cipher_suites_with_tls12_chacha20),
+                    .suites = cipher_suites_with_tls12_chacha20,
+                    .allow_chacha20_boosting = true
+            };
+
+            struct s2n_security_policy security_policy = {
+                .minimum_protocol_version = S2N_TLS12,
+                .cipher_preferences = &cipher_preferences,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+    
+            DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(connection);
+
+            connection->security_policy_override = &security_policy;
+            connection->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+            connection->client_protocol_version = S2N_TLS12;
+            connection->actual_protocol_version = S2N_TLS12;
+            connection->server_protocol_version = S2N_TLS12;
+
+            struct s2n_config *rsa_only_config;
+            EXPECT_NOT_NULL(rsa_only_config = s2n_config_new());
```

Comment:
Nit:

You can use DEFER_CLEANUP to avoid needing to call s2n_config_free later.
```suggestion
            DEFER_CLEANUP(struct s2n_config *rsa_only_config = s2n_config_new(), s2n_config_ptr_free);
            EXPECT_NOT_NULL(rsa_only_config);
```
I'm also not a fan of declaring a variable without initing it, even if it's set on the very next line. As a reviewer, the naked declaration catches my eye, and it forces me to keep reading to verify that there's no actual problem and the variable does eventually get initialized. We want our code to be obviously safe, and we want to minimize the effort required to verify its safety. This is an extremely minor example, but since there's no real benefit to NOT initializing variables when they're declared, I would suggest getting in that habit.